### PR TITLE
[Config] Update 'Audit log' to using a new config

### LINKED ISSUE #3667: [Config] Update 'Audit log' to using a new config
**Product:** Tarantool
**Since:** 3.0
**Root document:** 
- https://www.tarantool.io/en/enterprise_doc/appendix/audit_log/
- https://www.tarantool.io/en/doc/latest/reference/configuration/configuration_reference/
- example in https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/config/instances.enabled

**SME:** @ Totktonada

# Details

Environment variables:

```
% tarantool --help-env-list | grep AUDIT
| TT_AUDIT_LOG_EXTRACT_KEY                    | boolean        | false                                         | Enterprise Edition |
| TT_AUDIT_LOG_FILE                           | string         | var/log/{{ instance_name }}/audit.log         | Enterprise Edition |
| TT_AUDIT_LOG_FILTER                         | array          | nil                                           | Enterprise Edition |
| TT_AUDIT_LOG_FORMAT                         | string         | json                                          | Enterprise Edition |
| TT_AUDIT_LOG_NONBLOCK                       | boolean        | false                                         | Enterprise Edition |
| TT_AUDIT_LOG_PIPE                           | string         | box.NULL                                      | Enterprise Edition |
| TT_AUDIT_LOG_SPACES                         | array          | box.NULL                                      | Enterprise Edition |
| TT_AUDIT_LOG_SYSLOG_FACILITY                | string         | local7                                        | Enterprise Edition |
| TT_AUDIT_LOG_SYSLOG_IDENTITY                | string         | tarantool                                     | Enterprise Edition |
| TT_AUDIT_LOG_SYSLOG_SERVER                  | string         | box.NULL                                      | Enterprise Edition |
| TT_AUDIT_LOG_TO                             | string         | devnull                                       | Enterprise Edition |
```

Schema: https://github.com/tarantool/tarantool/blob/aab0096945f0e1561429f45f7297e5082b2ea0af/src/box/lua/config/instance_config.lua#L1791

Migration from 2.11 to 3.0: https://docs.google.com/spreadsheets/d/1_yEK9N_hBVWIga9mOk7zZzA6Zd3mAq7e5KZsnrbgj30/edit#gid=0

### PR DESCRIPTION
**Pull request by xuniq**
* Updated and restructured [Audit module](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/enterprise/audit_log/) page
* Added [audit_log](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/reference/configuration/configuration_reference/#audit-log) reference section:
  * [audit_log.extract_key](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/reference/configuration/configuration_reference/#configuration-reference-audit-extract-key), [audit_log.spaces](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/reference/configuration/configuration_reference/#confval-audit_log.spaces)  - new options
  * other options are moved from the box.cfg reference
* Documented new audit log [fields](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/enterprise/audit_log/#structure-of-audit-log-event) - uuid, severity
* Added a new [Severity level](https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/enterprise/audit_log/#severity-level) subsection to Custom events section (with examples)

* Fixed typo in the `tt search` description

Fixes https://github.com/tarantool/doc/issues/3667
Fixes https://github.com/tarantool/enterprise_doc/issues/258
Fixes https://github.com/tarantool/enterprise_doc/issues/257
Fixes https://github.com/tarantool/enterprise_doc/issues/221 - **also related to 2.11, don't forget to update this in 2.11 branch**
Fixes https://github.com/tarantool/enterprise_doc/issues/248

### CODE REVIEW

File: doc/enterprise/audit_log.rst [hunk 1]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
+..  _audit-log-event-groups:
 
 Event groups
-------------
+~~~~~~~~~~~~
 
-You can simplify working with audit log events by using built-in event groups.
-For example, you can set to record only events related to the enabling of the audit log,
+Built-in event groups are used to filter the event types that you want to audit.
+For example, you can set to record only authorization events,
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
For example, you can set to record only authorization events
```

File: doc/enterprise/audit_log.rst [hunk 2]
```diff
@@ -1,40 +1,253 @@
...
 
 Audit module
 ============
 
-The audit module available in Tarantool Enterprise Edition writes messages that record Tarantool events in plain text, CSV, or JSON format.
+**Example on GitHub**: `audit_log <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/config/instances.enabled/audit_log>`_
 
-It provides detailed reports of security-related activities and helps you find and
+The audit module available in Tarantool Enterprise Edition writes messages that record Tarantool events in plain text,
+CSV, or JSON format.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: Not sure that we should mention audit log formats in the first sentence. I'd rephrase the first sentence in a more general way, smth like this:

_The audit module available in Tarantool Enterprise Edition allows you to record various events occurred in ... Each event ..._

Also, listing all supported formats in multiple places makes documentation support harder. I'd try to keep such info in the reference only.

File: doc/enterprise/audit_log.rst [hunk 3]
```diff
@@ -1,40 +1,253 @@
...
+
+Read more: :ref:`Audit log configuration reference <configuration_reference_audit>`.
+
+..  _audit-log-enable:
+
+Enable audit logging
+~~~~~~~~~~~~~~~~~~~~
+
+To enable audit logging, define the log location using the
+:ref:`audit_log.to <configuration_reference_audit_to>` option in the ``audit_log`` section of the configuration file.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: Given that `audit_log.to` is a full option path, we don't need to tell `in the ``audit_log`` section` - the option name already contains the section name.

File: doc/enterprise/audit_log.rst [hunk 4]
```diff
@@ -1,40 +1,253 @@
...
+Possible log locations:
+
+*   a :ref:`file <configuration_reference_audit_file>`
+*   a :ref:`pipe <configuration_reference_audit_pipe>`
+*   a :ref:`system logger <configuration_reference_audit_syslog-server>`
+
+To disable audit logging, set the ``audit_log`` option to ``devnull``.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: Let's tell this after we show how to enable it. BTW, shouldn't it be `set the ``audit_log.to`` option` (add `to`)?

File: doc/enterprise/audit_log.rst [hunk 5]
```diff
@@ -1,40 +1,253 @@
...
+
+In the configuration below, the :ref:`audit_log.to <configuration_reference_audit_to>` option is set to ``file``.
+It means that the logs are written to a file.
+In this case, you should also define a file path (for example, ``audit_tarantool.log``) using
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: > (for example, ``audit_tarantool.log``)

I'd remove this - the example already shows this value.

File: doc/enterprise/audit_log.rst [hunk 6]
```diff
@@ -1,40 +1,253 @@
...
+the :ref:`audit_log.file <configuration_reference_audit_file>` option.
+If the option is omitted, the default path is used: ``var/log/instance001/audit.log``.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: It's. a bit weird that we show the default path right between the description on how to customize the path and the example. I'd move this earlier, for example: 

_By default, audit logs are saved... To specify the path to an audit log file explicitly ..._

Or we can move the default value after the example.

File: doc/enterprise/audit_log.rst [hunk 7]
```diff
@@ -1,40 +1,253 @@
...
+-   Filter by group. You can specify a list of :ref:`event groups <audit-log-event-groups>` to be recorded. For example,
+    select ``auth`` and ``priv`` to see the events related to authorization and granted privileges:
+
+    ..  code-block:: yaml
+
+        audit_log:
+          filter: [ auth,priv ]
+
+-   Filter by group and event. You can specify a group and a certain event depending on the purpose.
+    In the configuration below, ``user_create``, ``data_operations``, ``ddl``, and ``custom`` are selected to see the events related to
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
    In the configuration below, ``user_create``, ``data_operations``, ``ddl``, and ``custom`` are selected to see the events related to:
```

File: doc/enterprise/audit_log.rst [hunk 8]
```diff
@@ -1,40 +1,253 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the :ref:`audit_log.format <configuration_reference_audit_format>` option to choose the format of audit log events
+-- plain text, CSV, or JSON.
+
+JSON is used by default. It is more convenient to receive log events, analyze them, and integrate them with other systems if needed.
+The plain format can be efficiently compressed.
+The CSV format allows you to view audit log events in tabular form.
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/audit_log/config.yaml
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst:107]: I'd place the example before the use cases - right after the first sentence where we tell which option to use.

File: doc/enterprise/audit_log.rst [hunk 9]
```diff
@@ -1,40 +1,253 @@
...
+    :start-at: format:
+    :end-at: json
+    :dedent:
+
+..  _audit-log-spaces:
+
+Specify the spaces to be logged
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since :doc:`3.0.0 </release/3.0.0>`, :ref:`audit_log.spaces <configuration_reference_audit_spaces>` is used to specify
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: Let's drop all `Since ...` mentions from the topic. I think, this info is OK for What's new or option description in the reference page.

P.S. For field descriptions in `Structure of audit log event` having this info is OK because it is kind of a reference section.

File: doc/enterprise/audit_log.rst [hunk 10]
```diff
@@ -1,40 +1,253 @@
...
+
+..  _audit-log-run-read:
+
+Reading audit logs
+------------------
+
+In the example, the logs are written to the ``audit_tarantool.log`` file.
+Let's create a space ``bands`` and check the logs in the file after the creation:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst:162]: This code snippet shows five actions but the corresponding audit log entry is for `space_create` only. I believe, we can include this line only: `box.schema.space.create('Bands')`.

File: doc/enterprise/audit_log.rst [hunk 11]
```diff
@@ -1,40 +1,253 @@
...
+    :end-at: true
+    :dedent:
+
+..  _audit-log-run-read:
+
+Reading audit logs
+------------------
+
+In the example, the logs are written to the ``audit_tarantool.log`` file.
+Let's create a space ``bands`` and check the logs in the file after the creation:
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: > Let's create

Am I correct that we try to avoid first and second person writing?

https://www.tarantool.io/en/doc/latest/contributing/docs/style/#don-t-say-we

We can change to smth like this:

_In the example below, the `***` function creates the `***` ..._

File: doc/enterprise/audit_log.rst [hunk 12]
```diff
@@ -1,40 +1,253 @@
...
-audit different events for different reasons. Tarantool provides such an option for each of them.
+Then insert one tuple to space:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua
+    :language: lua
+    :start-after: load_data()
+    :end-before: box.space.bands:insert { 2
+    :dedent:
+
+As the ``extract_key`` option is set to ``true``, the audit system prints the primary key instead of the full tuple:
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: > As the ``extract_key`` option is set to ``true``

This was many sentences ago and a user might forget this fact. So, I'd change to `If the ``extract_key`` option ...`

File: doc/enterprise/audit_log.rst [hunk 13]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
 *   System events -- events related to modification or configuration of resources.
     For example, such logs record the replacement of a space.
 
-*   :ref:`User-defined events <audit-log-custom>`-- any events added manually using
+*   :ref:`Custom events <audit-log-custom>`-- any events added manually using
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
*   :ref:`Custom events <audit-log-custom>` -- any events added manually using
```

File: doc/enterprise/audit_log.rst [hunk 14]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
 
 ..  container:: table
 
     ..  list-table::
-        :widths: 30 35 35
+        :widths: 20 25 20 35
         :header-rows: 1
 
         *   -   Event
             -   Type of event written to the audit log
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
            -   Event type
```

File: doc/enterprise/audit_log.rst [hunk 15]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
+            -   Severity level
             -   Example of an event description
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
            -   Example
```

File: doc/enterprise/audit_log.rst [hunk 16]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
+
+..  container:: table
+
+    ..  list-table::
+        :widths: 30 35 35
+        :header-rows: 1
+
+        *   -   Field
+            -   Description
+            -   Example of a log field display
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]:
```suggestion
            -   Example
```

File: doc/enterprise/audit_log.rst [hunk 17]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
+        *   -   ``tag``
+            -   A text field that can be overwritten by the user
+            -
+        *   -   ``description``
+            -   Human-readable event description
+            -   ``Authenticate user Admin``
+
+    ..  warning::
+
+        You can set all these parameters only once.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: Hmmm.. Do you know what does this mean? Looks scary and unclear.

File: doc/enterprise/audit_log.rst [hunk 18]
```diff
@@ -45,115 +258,215 @@ decide whether you need to take actions:
...
+            -    Since :doc:`3.0.0 </release/3.0.0>`. A unique identifier of audit log event
+            -   ``cb44fb2b-5c1f-4c4b-8f93-1dd02a76cec0``
+        *   -   ``severity``
+            -   Since :doc:`3.0.0 </release/3.0.0>`. A severity level. Each system audit event has a severity level determined by its importance.
+                :ref:`Custom events <audit-log-custom>` have the ``INFO`` severity level by default.
+
+            -   ``VERBOSE``
+        *   -   ``remote``
+            -   Remote host that triggered the event
+            -   ``unix/:(socket)``
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst:427]: An IP address example might be more real-life but let's keep it as is. Just a random thought :)

File: doc/enterprise/audit_log.rst [hunk 19]
```diff
@@ -647,8 +634,8 @@ How many events can be recorded?
 
 If you write to a file, the size of the Tarantool audit module is limited by the disk space.
 If you write to a system logger, the size of the Tarantool audit module is limited by the system logger.
-If you write to a pipe, the size of the Tarantool audit module is limited by the system buffer
-if the ``audit_nonblock`` = ``false``; if ``audit_nonblock`` = ``true``, there is no limit.
+If you write to a pipe, the size of the Tarantool audit module is limited by the system buffer.
+If the ``audit_nonblock = false``, if ``audit_nonblock`` = ``true``, there is no limit.
```
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: audit_nonblock -> audit_log.nonblock

https://docs.d.tarantool.io/en/doc/gh-3667-audit-new/reference/configuration/configuration_reference/#configuration-reference-audit-nonblock

File: doc/reference/configuration/configuration_reference.rst [hunk 20]
```diff
@@ -11,6 +11,328 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+* :ref:`audit_log.syslog_server <configuration_reference_audit_syslog-server>`
+* :ref:`audit_log.to <configuration_reference_audit_to>`
+
+..  _configuration_reference_audit_extract_key:
+
+..  confval:: audit_log.extract_key
+
+    **Since:** :doc:`3.0.0 </release/3.0.0>`.
+
+    Specify the logging mode in DML events.
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]: >  Specify the logging mode in DML events.

I think, this sentence might be dropped as the next one tells exactly what this option does.

File: doc/reference/configuration/configuration_reference.rst [hunk 21]
```diff
@@ -11,6 +11,328 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    | Environment variable: TT_AUDIT_LOG_FILE
+
+..  _configuration_reference_audit_filter:
+
+..  confval:: audit_log.filter
+
+    Enable logging for a specified subset of audit events.
+    This option accepts the following values:
+
+    *   event names (for example, ``password_change``). For details, see :ref:`Audit log events <audit-log-events>`.
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]:
```suggestion
    *   Event names (for example, ``password_change``). For details, see :ref:`Audit log events <audit-log-events>`.
```

File: doc/reference/configuration/configuration_reference.rst [hunk 22]
```diff
@@ -11,6 +11,328 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    *   event groups (for example, ``audit``).  For details, see :ref:`Event groups <audit-log-event-groups>`.
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]:
```suggestion
    *   Event groups (for example, ``audit``).  For details, see :ref:`Event groups <audit-log-event-groups>`.
```

File: doc/reference/configuration/configuration_reference.rst [hunk 23]
```diff
@@ -11,6 +11,328 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    -   ``devnull``: disable audit logging.
+    -   ``file``: write audit logs to a file (see :ref:`audit_log.file <configuration_reference_audit_file>`).
+    -   ``pipe``: write audit logs to a pipe (see :ref:`audit_log.pipe <configuration_reference_audit_pipe>`).
+    -   ``syslog``: write audit logs to a system logger (see :ref:`audit_log.syslog <configuration_reference_audit_pipe>`).
+
+    By default, audit logging is disabled.
+
+    **Example**
+
+    The basic audit log configuration in the :doc:`3.0.0 </release/3.0.0>` version might look as follows:
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst]:
```suggestion
    The basic audit log configuration might look as follows:
```

File: doc/reference/configuration/configuration_reference.rst [hunk 24]
```diff
@@ -11,6 +11,328 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/audit_log/config.yaml
+        :language: yaml
```
Comment (andreyaksenov) [inline doc/reference/configuration/configuration_reference.rst:305]: Let's add `end-at: extract_key` to show only audit log config settings. I'd also add `start-at: audit_log` to make this work even if someone move the section.

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

File: doc/enterprise/audit_log.rst (hunk 6 above)
Comment (andreyaksenov) [inline doc/enterprise/audit_log.rst]: > the default path is used: ``var/log/instance001/audit.log``.

Please double check. Looks like its value should be dynamic: `var/log/{{ instance_name }}/audit.log`.

**Review by andreyaksenov — APPROVED** [review]
Great work! LGTM

File: doc/code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua [hunk 25]
```diff
@@ -0,0 +1,51 @@
+-- myapp.lua --
+
+-- Create space
+function create_space()
+    box.schema.space.create('Bands')
+    box.space.bands:format({
+        { name = 'id', type = 'unsigned' },
+        { name = 'band_name', type = 'string' },
+        { name = 'year', type = 'unsigned' }
+    })
```
Comment (Totktonada) [inline doc/code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua]: I guess it doesn't work: `Bands` and `bands` are different names. There is no `bands` space.

File: doc/reference/configuration/configuration_reference.rst [hunk 26]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
 Most of the configuration options described in this reference can be applied to a specific instance, replica set, group, or to all instances globally.
 To do so, you need to define the required option at the :ref:`specified level <configuration_scopes>`.
 
+..  _configuration_reference_audit:
+
+audit_log
+---------
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:17]: Shouldn't we keep the `config` section first?

File: doc/reference/configuration/configuration_reference.rst [hunk 27]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    *   Event names (for example, ``password_change``). For details, see :ref:`Audit log events <audit-log-events>`.
+    *   Event groups (for example, ``audit``).  For details, see :ref:`Event groups <audit-log-event-groups>`.
+
+    The option contains either one value from above or a combination of them.
+
+    To enable :ref:`user-defined audit log events <audit-log-custom>`, specify the ``custom`` value in this option.
+
+    |
+    | Type: array
+    | Possible values: 'all', 'audit', 'auth', 'priv', 'ddl', 'dml', 'data_operations', 'compatibility',
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:97]: Shouldn't we show that a possible value is, for example, `[audit, auth]`, not just `audit`? Now it looks like just one value can be set there.

One may also assume that the format is `audit,auth` -- a string of comma separated values.

Don't know, how to express it formally, but it seems, it should be clear from the reference documentation.

Comment (Totktonada) [reply doc/reference/configuration/configuration_reference.rst:97]: There are examples for another options. I would add one here too.

File: doc/reference/configuration/configuration_reference.rst [hunk 28]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    | Default: false
+    | Environment variable: TT_AUDIT_LOG_NONBLOCK
+
+..  _configuration_reference_audit_pipe:
+
+..  confval:: audit_log.pipe
+
+    Specify a pipe for the audit log destination.
+    You can set the ``pipe`` type using the :ref:`audit_log.to <configuration_reference_audit_to>` option.
+    If log is a program, its pid is stored in the ``audit_log.logger_pid`` variable.
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: Is `audit_log.logger_pid` means `audit_log` module (obtained by `require('audit_log')`) and its field `logger_pid`?

Now it looks very similar to `audit_log.to` and others, which are configuration options in this context. Maybe it worth to clarify it here to avoid a confusion.

File: doc/reference/configuration/configuration_reference.rst [hunk 29]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/audit_log_pipe/config.yaml
+        :language: yaml
+        :start-at: audit_log:
+        :end-at: '| cronolog audit_tarantool.log'
+        :dedent:
+
+    This starts the `cronolog <https://linux.die.net/man/1/cronolog>`_ program when the server starts
+    and sends all ``audit_log`` messages to cronolog standard input (``stdin``).
+    If the ``audit_log`` string starts with '|',
+    the string is interpreted as a Unix `pipeline <https://en.wikipedia.org/wiki/Pipeline_%28Unix%29>`_.
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:179]: A pipeline connected to process'es stdin, I guess?

So, if there is no pipe at start, where the logs will be send?

(These my questions as a reader, I don't remember how it works.)

File: doc/reference/configuration/configuration_reference.rst [hunk 30]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    | Type: string
+    | Possible values: 'auth', 'authpriv', 'cron', 'daemon', 'ftp', 'kern', 'lpr', 'mail', 'news', 'security', 'syslog', 'user', 'uucp', 'local0', 'local1', 'local2', 'local3', 'local4', 'local5', 'local6', 'local7'
+    | Default: 'local7'
+    | Environment variable: TT_AUDIT_LOG_SYSLOG_FACILITY
+
+..  _configuration_reference_audit_syslog-identity:
+
+..  confval:: audit_log.syslog_identity
+
+    Specify an arbitrary string that will be placed at the beginning of all messages.
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: > <...> string that will be placed at the beginning of all messages.

This is not exactly so.

I would better say what does it means: an application name to show in logs.

There is a brief description of the syslog format [here](https://github.com/tarantool/tarantool/blob/42d0a7a834ac75fb696d116379b00ce107c0ad1d/src/lib/core/say.c#L987-L999).

File: doc/reference/configuration/configuration_reference.rst [hunk 31]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+    -   :ref:`audit_log.syslog_facility <configuration_reference_audit_syslog-facility>` -- a system logger keyword that tells syslogd where to send the message.
+        The default value is ``local7``.
+
+    -   :ref:`audit_log.syslog_identity <configuration_reference_audit_syslog-identity>` -- a string placed at the beginning of every message.
+        The default value is ``tarantool``.
+
+    These options are interpreted as a message for the `syslogd <https://datatracker.ietf.org/doc/html/rfc5424>`_ program,
+    which runs in the background of any Unix-like platform.
+
+    An example of a Tarantool audit log entry in the syslog:
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:268]: Hm. The example is JSON, not in the syslog format. But this sentence states that it is from syslog.

We can write JSON messages to syslog (AFAIR, they shouldn't contain newline symbols in this case) and they will be formatted with all the proper headers and then the JSON data on the same line.

File: doc/reference/configuration/configuration_reference.rst [hunk 32]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+..  _configuration_reference_audit_to:
+
+..  confval:: audit_log.to
+
+    Enable audit logging and define the log location.
+    This option accepts the following values:
+
+    -   ``devnull``: disable audit logging.
+    -   ``file``: write audit logs to a file (see :ref:`audit_log.file <configuration_reference_audit_file>`).
+    -   ``pipe``: write audit logs to a pipe (see :ref:`audit_log.pipe <configuration_reference_audit_pipe>`).
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst]: I would say 'start a program and pipe audit log to it' or 'start a program and write audit logs to it'. Just 'write to a pipe' doesn't add much to the option name.

File: doc/reference/configuration/configuration_reference.rst [hunk 33]
```diff
@@ -11,6 +11,329 @@ This topic describes all :ref:`configuration parameters <configuration>` provide
...
+        To do this, create separate paths and specify them.
+
+    |
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_AUDIT_LOG_SYSLOG_SERVER
+
+..  _configuration_reference_audit_to:
+
+..  confval:: audit_log.to
```
Comment (Totktonada) [inline doc/reference/configuration/configuration_reference.rst:288]: I guess that it is the most commonly used option, but it is the last alphabetically.

It is expected that a reference contains the options in the alphabetical order. Maybe we can highlight this option somehow differently?

I've no certain proposal, just thinking around.

Comment (Totktonada) [reply doc/reference/configuration/configuration_reference.rst:288]: The section starts with a link to the 'audit module' article, which is more like a guide. And it describes `audit_log.to` first. It seems, my concern is already solved.

File: doc/reference/configuration/configuration_reference.rst (hunk 26 above)
Comment (Totktonada) [reply doc/reference/configuration/configuration_reference.rst:17]: Since it is a reference, it seems logical to keep the alphabetical order.

See also [another comment](https://github.com/tarantool/doc/pull/3983#discussion_r1480284197) on the topic.

File: doc/enterprise/audit_log.rst [hunk 34]
```diff
@@ -647,8 +619,8 @@ How many events can be recorded?
 
 If you write to a file, the size of the Tarantool audit module is limited by the disk space.
```
Comment (Totktonada) [inline doc/enterprise/audit_log.rst]: Audit module -> audit log?

Comment (Totktonada) [reply doc/enterprise/audit_log.rst]: And the same below.

File: doc/enterprise/audit_log.rst [hunk 35]
```diff
@@ -647,8 +619,8 @@ How many events can be recorded?
...
 If you write to a system logger, the size of the Tarantool audit module is limited by the system logger.
-If you write to a pipe, the size of the Tarantool audit module is limited by the system buffer
-if the ``audit_nonblock`` = ``false``; if ``audit_nonblock`` = ``true``, there is no limit.
+If you write to a pipe, the size of the Tarantool audit module is limited by the system buffer.
+If the ``audit_log.nonblock = false``, if ``audit_log.nonblock`` = ``true``, there is no limit.
```
Comment (Totktonada) [inline doc/enterprise/audit_log.rst]: I guess, audit module -> audit message.

File: doc/enterprise/audit_log.rst [hunk 36]
```diff
@@ -647,8 +619,8 @@ How many events can be recorded?
...
 However, it is not recommended to use the entire memory, as this may cause performance degradation
 and even loss of some logs.
```
Comment (Totktonada) [inline doc/enterprise/audit_log.rst]: That's an unclear recommendation, because there are no configurable memory limit for the audit log.

**Review by Totktonada — APPROVED** [review]
No objections from my side. I left several questions to discuss if you like.

File: doc/code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua (hunk 25 above)
Comment (xuniq) [reply doc/code_snippets/snippets/config/instances.enabled/audit_log/myapp.lua]: Fixed

File: doc/enterprise/audit_log.rst (hunk 34 above)
Comment (xuniq) [reply doc/enterprise/audit_log.rst]: Fixed

File: doc/enterprise/audit_log.rst (hunk 35 above)
Comment (xuniq) [reply doc/enterprise/audit_log.rst]: Fixed

File: doc/reference/configuration/configuration_reference.rst (hunk 32 above)
Comment (xuniq) [reply doc/reference/configuration/configuration_reference.rst]: I agree. Thanks, fixed.

File: doc/reference/configuration/configuration_reference.rst (hunk 27 above)
Comment (xuniq) [reply doc/reference/configuration/configuration_reference.rst:97]: Added the example, updated the description a little.

File: doc/reference/configuration/configuration_reference.rst (hunk 30 above)
Comment (xuniq) [reply doc/reference/configuration/configuration_reference.rst]: Thanks, updated description
